File: backend/app/Models/Student.php
<?php

namespace App\Models;

use App\Enum\SchoolGradeEnum;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Student extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'name',
        'email',
        'school_id',
        'school_grade',
    ];

    protected $casts = [
        'school_grade' => SchoolGradeEnum::class,
    ];

    public function school()
    {
        return $this->belongsTo(School::class);
    }

    public function projects()
    {
        return $this->belongsToMany(Project::class, 'student_projects');
    }
}

File: backend-fastapi/app/models/student.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base
from app.enums.school_grade import SchoolGrade

class Student(Base):
    __tablename__ = "students"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False)
    email = Column(String(255), nullable=True)
    school_grade = Column(Integer, nullable=False)  # SchoolGrade enum
    school_id = Column(Integer, ForeignKey("schools.id"), nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    deleted_at = Column(DateTime(timezone=True), nullable=True)
    
    # Relationships
    school = relationship("School", back_populates="students")
    projects = relationship("Project", secondary="student_projects", back_populates="students") 